### DISCUSSION TITLE
Enabling JSON routes in production (to mirror dev behavior)

### DISCUSSION THREAD

**Discussion by jdmcd** [remix-run/remix#9708] [Q&A] [answered]
Hi all,

In development the data from my loaders is fetched via a route that returns JSON and not HTML. Remix seems to handle this automatically (i.e. something like `/my-route?_data=routes%2Fchallenge.2`). In production, the loader data appears to be be entirely server side rendered and injected directly into the HTML. 

Is there a way to enable the same JSON functionality inside of the production build? I'm not sure what feature this is inside of Remix so it's hard to search. 

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
The json data is inlined in the html during SSR for a document request which happens on the first page load.

On any client side navigation Remix will fetch only the data on the URL you mentioned.

This behavior remains the change in development and production. If you don’t see the fetching for data ensure your app has client side JS working on production.

**Reply by jdmcd** [reply]
Thanks - that's not what I'm seeing which is why I think I'm confused. How do I check if client side JS is enabled?

**Reply by jdmcd** [reply]
Is there any way to force the JSON route even on a full page refresh? (Instead of it being embedded via SSR)

**Reply by jdmcd** [reply]
Ah - I see this behavior now. So yeah the big question for me then is whether I can force one route in particular to always load its data via json instead of inlining via SSR

**Reply by jdmcd** [reply]
Ended up solving this by switching to a client loader - thanks for the help!
